Boyer-Moore/BoyerMoore.swift

```diff
@@ -6,7 +6,7 @@
   http://www.drdobbs.com/database/faster-string-searches/184408171
 */
 extension String {
-    func indexOf(pattern: String) -> String.Index? {
+    func indexOf(pattern: String, useHorspoolImprovement: Bool = false) -> String.Index? {
```

Comment:
I think `index(of pattern: String, useHorspoolImprovement: Bool = false) -> Index?` is a better method signature. 

Since Swift 3, many methods switched over from `indexOf` to `index(of:)`. `String.Index` is unnecessary since we're inside a `String` extension and the compiler can infer the type of `Index`.